packages/inter-protocol/src/provisionPoolKit.js

```diff
@@ -217,23 +232,32 @@ export const prepareProvisionPoolKit = (
           const refs = await deeplyFulfilledObject(obj);
           Object.assign(this.state, refs);
         },
+        /** @returns {import('@agoric/vats').BridgeHandler} */
         makeHandler() {
           const { bankManager, namesByAddressAdmin, walletFactory } =
             this.state;
           if (!bankManager || !namesByAddressAdmin || !walletFactory) {
             throw Fail`must set references before handling requests`;
           }
-          const { helper } = this.facets;
-          // a bit obtuse but leave for backwards compatibility with tests
-          const innerMaker = makeBridgeProvisionTool(
-            bank => helper.sendInitialPayment(bank),
-            () => helper.onProvisioned(),
-          );
-          return innerMaker({
+
+          const ephemera = provisionPoolEphemera(ephemeralToolsHandle);
+          if (!ephemera.makeBridgeProvisionTool) {
+            throw Fail`makeBridgeProvisionTool must be set to ephemera first`;
+          }
+
+          trace({
             bankManager,
             namesByAddressAdmin,
             walletFactory,
           });
+
+          const provisionHandler = ephemera.makeBridgeProvisionTool(
+            bankManager,
+            walletFactory,
+            namesByAddressAdmin,
+          );
```

Comment:
> The old code called makeBridgeProvisionTool() on every call of makeHandler(). Why does this switch to calling it only once per incarnation of the ProvisionPool?

Could you please take another look? `makeBridgeProvisionTool` is the maker returned from `prepareBridgeProvisionPool` so as far as I can see we are invoking the maker every time there's a call to `makeHandler`.

> I don't think we need ephemera at all. (I might be wrong. Tell me why you think we want it.)

> I don't see the value in using ephemera to cache the handlers. Was there some other reason to use ephemera?

The reason why we use ephemera is because of the `helper.sendInitialPayment` and `helper.onProvisioned` callbacks. We have to pass them like `prepareBridgeProvisionTool(zone, helper.sendInitialPayment, helper.onProvisioned)` because these specific functions are not durable(the `helper` facet is, but specific callbacks are not).  We create the maker in `helper.start` and create the ephemera like below.

https://github.com/Agoric/agoric-sdk/blob/a8b124a78d467f678caa903bf9b71ef00a13f76f/packages/inter-protocol/src/provisionPoolKit.js#L480-L487

If we try to pass those callbacks like:
```js
makeBridgeProvisionTool(
  bankManager, 
  walletFactory, 
  namesByAddressAdmin, 
  helper.sendInitialPayment, 
  helper.onProvisioned
)
```
Wwe get an error like `Remotables must be explicitly declared`.
 > Also tried `makeBridgeProvisionTool(bankManager, walletFactory, namesByAddressAdmin, bank => helper.sendInitialPayment(bank), () => helper.onProvisioned())`

I also tried;
```js
makeBridgeProvisionTool(
  bankManager, 
  walletFactory, 
  namesByAddressAdmin, 
  Far('helpers', { 
    sendInitialPayment: bank => helper.sendInitialPayment(bank), 
    onProvisioned: () => helper.onProvisioned()
  }
)
```

This time the exo package could serialize `Far('helpers')` but complained that it's not durable.

So the only solution I could think of was to;
* Define (call `prepareBridgeProvisioningTool`) bridge handler exo in `helper.start`
  * Make sure `prepareBridgeProvisioningTool` is fed with `helper.sendInitialPayment` and `helper.onProvisioned`
* In `helper.start` set the maker into the ephemera like: `ephemera.makeBridgeProvisionTool = makeBridgeProvisionTool;`
* In `makeHandler` consume references (written by `setReferences`) and `ephemera.makeBridgeProvisionTool`
* Use `ephemera.makeBridgeProvisionTool` to _instantiate_ the handler exo

### Why not define the exo in `makeHandler`?
SwingSet needs every exo object defined in the previous incarnation to be defined again during the start up phase. So defining it in `makeHandler` does not work since it is not invoked during the startup (needs `setReferences` to be called first).

### Why ephemera?
Because the properties of `this.state` in the `provisionPoolKit` are immutably defined during the first incarnation. So it is not possible to assign the maker for bridgeProvisioningHandler to the exo state like `this.state.makeBridgeProvisionHandler = makeBridgeProvisionHandler`. As a result we use ephemera to keep the reference to `makeBridgeProvisionHandler`.

Reply:
resolved in https://github.com/Agoric/agoric-sdk/commit/a7d00fc6fcfa78e7b88d42e370670d647652c4f6